test/python/circuit/test_parameters.py

```diff
@@ -555,21 +555,21 @@ def test_partial_assignment(self):
 
         circ = circ.assign_parameters({alpha: freq - delta})
         cal_sched = list(circ.calibrations['custom'].values())[0]
-        self.assertEqual(cal_sched.instructions[0][1].frequency, freq - delta + beta)
+        self.assertEqual(cal_sched.instructions[0].frequency, freq - delta + beta)
```

Comment:
Is the old `instructions` method being deprecated? Is it specifically for `ScheduleBlock`s? This looks like a breaking change to me

Reply:
This is only for `ScheduleBlock` since it doesn't have t0.

Reply:
This change would break user code, so it's a breaking interface change. Whenever possible, we should avoid this.

Perhaps for `ScheduleBlock.instructions`, we should try to convert to a Schedule, or print a warning, or error and ask users to use a different method. I'm not sure what the best, but we shouldn't expect them to make this change without some assistance

Reply:
I agree we want to avoid breaking changes when possible. Ideally, we convert to a `Schedule` as @lcapelluto describes and warn that this method will be deprecated in favour of `timed_instructions` in a future PR and use another property (what should this be named?) instead that does not raise this deprecation warning for this behavior. We could always reintroduce `instructions` as an attribute after the deprecation period.

Reply:
Thanks. I updated the behavior of `ScheduleBlock.instructions`. Now it converts `self` into `Schedule` and call `.instructions`. The original method is renamed to `ScheduleBlock.blocks`. I don't think this change has impact to the user code. I'm not sure if we need to rename `instructions` to `timed_instructions`, since this causes tons of deprecation warnings in tutorials.